### PR TITLE
Refactor hooks and simplify button

### PR DESCRIPTION
**Pull request by anxolin**
Follow up on https://github.com/mevblocker/web/pull/63

Refactored the hooks for the connect button and simplified the logic.

### CODE REVIEW

File: src/components/AddRpcButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  width: 100%;
+  margin: 2.4rem 0 0;
+  background: ${({ state }) => state === 'added' ? transparentize(0.8, Color.green) : transparentize(0.9, Color.orange)};
+  padding: 1rem;
+  border-radius: 1.2rem;
+  text-align: center;
+`
+
+export function AddRpcButton() {  
+  const { addWalletState, connectAndAddToWallet } = useConnectAndAddToWallet()
```
Comment (anxolin) [inline src/components/AddRpcButton/index.tsx:28]: way smaller logic for the component

File: src/components/AddRpcButton/hooks/useConnectAndAddToWallet.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    setAddRpcPromise(null)
+  }, [setState])
+  
+
+  // Add RPC endpoint to wallet (with analytics + handle timeout state)
+  const addToWallet = useAddRpcWithTimeout({
+    isAdding: state === 'adding',
+    addingPromise,
+    onAdding (newAddRpcPromise) {
+      console.debug('[connectAndAddToWallet] Adding RPC...')
```
Comment (anxolin) [inline src/components/AddRpcButton/hooks/useConnectAndAddToWallet.ts:50]: Allow me to leave these debug comments :) 

They don't show up in the console unless you enable the VERBOSE mode

**Review by shoom3301 — APPROVED** [review]
Looks great! Thanks

File: src/components/AddRpcButton/hooks/useAddRpcWithTimeout.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { useCallback } from "react";
+import { addRpcUrlAction as sendToAnalytics, AddRpcUrlActionType } from "@src/lib/analytics/events";
+import { ERROR_ADD_MANUALLY_MESSAGE } from "@src/utils/handleRpcError";
+import { useAddRpcEndpoint } from "@src/hooks/useAddRpcEndpoint";
+import { WalletClient } from "viem";
+import { AddToWalletStateValues } from "..";
```
Comment (alfetopito) [inline src/components/AddRpcButton/hooks/useAddRpcWithTimeout.ts:6]: Do we use `;` in this project? Or is it just the editor adding it because it like it?

**Review by alfetopito — APPROVED** [review]

File: src/components/AddRpcButton/hooks/useAddRpcWithTimeout.ts (hunk 3 above)
Comment (anxolin) [reply src/components/AddRpcButton/hooks/useAddRpcWithTimeout.ts:6]: I think the lint setup has them. We could remove, i don't like them either

**Pull request merged**